homeassistant/components/spanet/water_heater.py

```diff
@@ -0,0 +1,256 @@
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the spa entity."""
+    access_token = str(entry.data.get("access_token"))
+    refresh_token = str(entry.data.get("refresh_token"))
+    name = str(entry.data.get("spa_name"))
+
+    spa = Spa(name, access_token, refresh_token)
+    async_add_entities([spa])
+
+
+class Spa(WaterHeaterEntity):
+    """SpaNET Spa water heater."""
+
+    def __init__(self, name: str, access_token: str, refresh_token: str) -> None:
+        """Initialize SpaNET Spa water heater."""
+        self._name = name
+        self._access_token = access_token
+        self._refresh_token = refresh_token
+        self._current_temperature = 0.0
+        self._target_temperature = 0.0
+        self._is_away_mode_on = False
+        self._operation_mode = ""
+
+    @property
+    def name(self) -> str:
+        """Return the name of the spa."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return the name of the spa water heater."""
+        return f"{DOMAIN}_{self._name}"
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the target temperature."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 10.0
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return 41.8
+
+    @property
+    def is_away_mode_on(self) -> bool:
+        """Return the status of AWAY MODE."""
+        return self._is_away_mode_on
+
+    @property
+    def current_operation(self) -> str:
+        """Return the current Operation Mode."""
+        return self._operation_mode
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement for temperature."""
+        return "°C"
+
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
+        """Set the current Operation Mode - NORM, AWAY, WEEKEND."""
+        await self.setOperationModeAPI(operation_mode)
+
+    async def async_update(self) -> None:
+        """Retrieve updated data by calling the API."""
+        self._current_temperature = await self.getCurrentTemperatureAPI()
+        self._target_temperature = await self.getTargetTemperatureAPI()
+
+        self._operation_mode = await self.getOperationModeAPI()
+        self._is_away_mode_on = self._operation_mode.lower() == "AWAY"
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set temperature of spa."""
+        if "temperature" in kwargs:
+            await self.setTemperatureAPI(int(kwargs["temperature"] * 10))
+
+    async def getCurrentTemperatureAPI(self) -> float:
```

Comment:
We don't allow communication directly with the devices/services and those calls should be made by a library hosted on pypi.
See https://developers.home-assistant.io/docs/creating_component_code_review/#4-communication-with-devicesservices

Reply:
Read PR description. Still missing set requirements in manifest and other files etc.
Putting this back in draft until all requirements is fixed.
Thanks

Reply:
Hi, thanks for reviewing my pull request. I've removed those calls from the water_heater.py file and used a pypi package instead. It's in commit 'eef785728447e38abcac1a83bf978b2aa39b43b3' made a couple of days ago. Please let me know if I misunderstood what is required and I'll fix it ASAP. Thanks

Reply:
You have not read and filled the PR description as documentation is missing, manifest is missing your dependency e.g. the package you made into pypi etc.

Reply:
Hi, my apologies. Where do I add the documentation for this: Documentation added/updated for [www.home-assistant.io](https://github.com/home-assistant/home-assistant.io)?

Reply:
" Link to documentation pull request: "

Reply:
Hi, I've updated my manifest files under requirements for my pypi package but I'm not sure where to add documentation. Could you please redirect me to the right direction. Thanks